web-console/knikubevirt/snapshots/snapshots.md

```diff
@@ -0,0 +1,72 @@
+# Virtual Machines Snapshots
+
+Snapshots allow users to create a copy of the VM disks and memory, mainly for restore and backup purposes. The snapshots are taken live or offline depending on if the VM is running or not. 
+This document describes the snapshot design flow.  
+
+
+
+
+## From VM List 
+
+![Offline VM list - snapshot action from a single VM kebab](img/1-00.jpg)
+When clicking on the kebab menu in the VM list, the user will be able to take a snapshot of the VM, meaning he will copy the disks, memory content, and metadata for restoring, backup, creating templates and cloning VMs.
+
+![Offline Snapshot modal - default](img/1-01.jpg)
+In the basic modal, the snapshot will take all current configuration and selected parameters.
+The user must add a descriptive name.
+![Offline Snapshot modal - selecting disks](img/1-02.jpg)
+In the disks field, the user will be able to decide if to include all disks or just some of them and if to include memory as part of the snapshot or not.
+![Offline Snapshot modal - disks customization](img/1-03.jpg)
+![Offline Snapshot modal - disks customized](img/1-04.jpg)
+
+![Offline Snapshotting status indication](img/1-05.jpg)
+![Offline Snapshotting status indication on click and kebab mode](img/1-06.jpg)
+
+Any virtual machine that has a snapshot in progress will show this in the state:
+When in this progress mode all actions would be disabled and the “Stop snapshot” would be available.
+
+
+## VM page - snapshots tab
+
+![Offline Shanpshot tab](img/2-00.jpg)
+The snapshot list is located under the VM. A tab item in the horizontal navigation will be dedicated for snapshots. The list is ordered by the creation date as a default. 
+The user can sort the list by name or date, clicking on the title will allow the user to switch sorting and decide if it's ascending or descending.  
+![Offline Snapshot - same modal as in VM list](img/2-01.jpg)
+Same modal as in VM list - Snapshot action
+![Offline Snapshot tab - expanded item](img/2-02.jpg)
```

Comment:
FYI @matthewcarleton we do not yet have the expanded table today in the console so worth syncing with @priley86 regarding the PF4 conversion for this table.

Reply:
@beanh66 I know that @yuvalgalanti is working with a developer to potentially contribute this to PF4. @yuvalgalanti maybe it's worth syncing with @priley86 after to see if there is anything specific needed to apply to pull it into OpenShift.

Reply:
talking with @priley86 it sounds like this expanding table row is a possibility but might need a larger group conversation around that.

Reply:
Let me know if you want to remove the drawer from list item and view more details on a modal, for now.

Reply:
I may just be out of the loop, but if the snapshot's name is blue and clickable here then that would mean that a Snapshot Details page would exist, right?

That Details page would probably include all of the same information shown in this dropdown region. Is there a use case for being able to see all of this information without going into the snapshot's Details page? Should the snapshot's name not actually be a clickable blue in these screens?

I'm also not clear on what the Disks, Pods, Networks, and Applications tabs would include here. I assume those resources wouldn't be shown in a table because then we'd have a table within a table, and the resources could have been changed or deleted since the time the snapshot was taken so I also assume they'd just be static names of disks/NICs/etc. If that's the case, does that information need to be spread across multiple tabs? What would the tabs look like in the snapshot's Details view?

Reply:
I'm leaning more towards the simplicity of the expanded row without a details view. It feels like it's more valuable to quickly see the details of the snapshot and not be required to go into a detail page. 
It might be worthwhile to pull @yuvalgalanti into this conversation for some background on how he landed on the details that is being shown here.

Reply:
@matthewcarleton This might come down to whether or not a "Snapshot" is actually an object as it's represented here with a blue resource badge. If it is, does that mean it would also have a YAML tab? If yes, then a Details page seems kinda inevitable. If it's not an object and doesn't have YAML, then maybe a "View details" or "View info" modal as @Ranelim suggested above is an easier short-term solution.

Reply:
+1 I think that's a good reason to pull in @yuvalgalanti to shed some light on the decisions made here. If it's a gap we can create an issue and update the design. Then this PR doesn't grow beyond a reasonable scope.

Reply:
Good point @andybraren, we should find out if the drill down is necessary. If not, the expand section makes sense. If we have to use the drill down page, maybe the expand section isn't as important. I'll leave that to you @Ranelim!

Reply:
I think we better go with the modal for now. We already have it, it does not take the user out of context and it's lite. I agree that a drawer is a better solution here, so maybe we'll update to a drawer once it is an official component in PF4.

Now that I look at it again, I find it weird that there is one view of snapshot detail and a shorter version of the snapshot details just before restoring it... So, we may want to combine them